src/main/kotlin/com/ampnet/payoutservice/controller/PayoutController.kt

```diff
@@ -2,36 +2,55 @@ package com.ampnet.payoutservice.controller
 
 import com.ampnet.payoutservice.controller.request.CreatePayoutRequest
 import com.ampnet.payoutservice.controller.response.CreatePayoutResponse
-import com.ampnet.payoutservice.service.PayoutService
+import com.ampnet.payoutservice.controller.response.CreatePayoutTaskResponse
+import com.ampnet.payoutservice.exception.ErrorCode
+import com.ampnet.payoutservice.exception.ResourceNotFoundException
+import com.ampnet.payoutservice.model.params.CreatePayoutTaskParams
+import com.ampnet.payoutservice.service.CreatePayoutQueueService
 import com.ampnet.payoutservice.util.BlockNumber
 import com.ampnet.payoutservice.util.ChainId
 import com.ampnet.payoutservice.util.ContractAddress
 import com.ampnet.payoutservice.util.WalletAddress
 import org.springframework.http.ResponseEntity
 import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RestController
+import java.util.UUID
 
 @RestController
-class PayoutController(private val payoutService: PayoutService) {
+class PayoutController(val createPayoutQueueService: CreatePayoutQueueService) {
 
-    @PostMapping("/payout/{chainId}/{assetAddress}/create")
+    @GetMapping("/payouts/tasks/{taskId}")
+    fun getPayoutTaskById(@PathVariable taskId: UUID): ResponseEntity<CreatePayoutTaskResponse> {
+        return createPayoutQueueService.getTaskById(taskId)
+            ?.let { ResponseEntity.ok(it) }
+            ?: throw ResourceNotFoundException(
+                ErrorCode.PAYOUT_TASK_NOT_FOUND,
+                "Create payout task not found"
+            )
+    }
+
```

Comment:
https://linear.app/ampnet/issue/SD-556/basic-payout

Looks like there should be a route to get all payouts for a specific issuer, screen 1.

Reply:
Yes, this will be handled in https://linear.app/ampnet/issue/SD-572/add-endpoints-to-fetch-all-payouts-per-issuer-and-investor